src/LocaleProvider.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\I18n;
+
+/**
+ * LocaleProvider is a stateful service that stores current locale.
+ * Other services may use it as a dependency.
+ */
+final class LocaleProvider
+{
+    private ?Locale $locale = null;
+
+    public function __construct(
+        private Locale $defaultLocale
+    ) {
+    }
+
+    public function get(): Locale
```

Comment:
Is there any use cases when a developer need to know was locale changed from default?

Reply:
Kind of `wasChanged(): bool => $this->locale !== null`

Reply:
Yes. For the purpose we have `getDefaultLocale()`.